pkg/sources/git/git.go

```diff
@@ -522,7 +522,26 @@ func (s *Git) ScanCommits(ctx context.Context, repo *git.Repository, path string
 		repoCtx = context.WithValue(ctx, "repo", path)
 	}
 
-	diffChan, err := s.parser.RepoPath(repoCtx, path, scanOptions.HeadHash, scanOptions.BaseHash == "", scanOptions.ExcludeGlobs, scanOptions.Bare)
+	logger := repoCtx.Logger()
+	var (
+		logValues []any
+
+		baseHashSet bool
```

Comment:
Added these so we aren't doing string compares for every single diff.

Reply:
This looks like it only saves a comparison when the base hash isn't set - and in that case, isn't the comparison constant-time (and fast) anyway?

Reply:
It's also used in the call to `RepoPath`, and similar to below, it felt wrong to do the same comparison for every diff, when we only need to do it once.